BrowserStack.java

```diff
@@ -34,36 +34,83 @@ public static void myassert(boolean  x) {
             throw new IllegalArgumentException("Assert fail") ;
         }
     }
+    
     //push() - pushes the element in this case its a url (String)
     private  void push(String url){
-        //WRITE YOUR CODE HERE
+        top++;
+        myStack[top] = url;
     }
+
     //pop()- pops the element from the stack
     private String pop(){
-        //WRITE YOUR CODE HERE
+        String popElement = myStack[top];
```

Comment:
Memory leak issue. Though you decrement the pointer. The old value still resides in the stack until you override with a new value.

`` myStack[top] = null; ``
will impress the interviewer.